source/fe/fe_poly_tensor.cc

```diff
@@ -446,56 +423,54 @@ FE_PolyTensor<dim, spacedim>::fill_fe_values(
                   for (unsigned int k = 0; k < n_q_points; ++k)
                     for (unsigned int d = 0; d < dim; ++d)
                       output_data.shape_values(first + d, k) =
-                        fe_data.shape_values[i][k][d];
+                        fe_data.shape_values[dof_index][k][d];
                   break;
                 }
 
               case mapping_covariant:
               case mapping_contravariant:
                 {
-                  mapping.transform(make_array_view(fe_data.shape_values, i),
-                                    mapping_kind,
-                                    mapping_internal,
-                                    make_array_view(
-                                      fe_data.transformed_shape_values));
+                  mapping.transform(
+                    make_array_view(fe_data.shape_values, dof_index),
+                    mapping_kind,
+                    mapping_internal,
+                    make_array_view(fe_data.transformed_shape_values));
 
                   for (unsigned int k = 0; k < n_q_points; ++k)
                     for (unsigned int d = 0; d < dim; ++d)
                       output_data.shape_values(first + d, k) =
-                        fe_data.transformed_shape_values[k][d];
+                        dof_sign * fe_data.transformed_shape_values[k][d];
 
                   break;
                 }
 
               case mapping_raviart_thomas:
               case mapping_piola:
                 {
-                  mapping.transform(make_array_view(fe_data.shape_values, i),
-                                    mapping_piola,
-                                    mapping_internal,
-                                    make_array_view(
-                                      fe_data.transformed_shape_values));
+                  mapping.transform(
+                    make_array_view(fe_data.shape_values, dof_index),
+                    mapping_piola,
+                    mapping_internal,
+                    make_array_view(fe_data.transformed_shape_values));
                   for (unsigned int k = 0; k < n_q_points; ++k)
                     for (unsigned int d = 0; d < dim; ++d)
                       output_data.shape_values(first + d, k) =
-                        fe_data.sign_change[i] *
-                        fe_data.transformed_shape_values[k][d];
+                        dof_sign * fe_data.transformed_shape_values[k][d];
```

Comment:
I would not touch Nedelec elements here. I think the issue with non-standard meshes should either be completely resolved in a separate PR or we do not touch them at all.

Reply:
Nedelec is not touched here at all. All dof signes stay at +1 except for elements that re-implement a certain table.